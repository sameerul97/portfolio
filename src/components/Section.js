import React from 'react'

/* Params: {id, classes} classes are optional */
export default function Section({ id, classes, children }) {
  return (
    <section className={`utils ${classes ? classes : ''}`} id={id}>
      <Board>{children}</Board>
    </section>
  )
}

const Board = ({ children }) => <div className="m-auto rounded board">{children}</div>
