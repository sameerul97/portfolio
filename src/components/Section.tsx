import React, { ReactNode } from 'react'

import { SectionProps } from '../interfaces'

/* Params: {id, classes} Classes param accepts string. I.e override section bg color with bg-red */
export default function Section({ id, classes, children }: SectionProps) {
  return (
    <section className={`utils ${classes ? classes : ''}`} id={id ? id : ''}>
      <Board>{children}</Board>
    </section>
  )
}

const Board = ({ children }: { children: ReactNode }) => <div className="m-auto rounded board">{children}</div>

export { Board }
