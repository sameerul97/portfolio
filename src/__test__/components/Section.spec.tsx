import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { screen } from '@testing-library/react'

/* Components */
import Section, { Board } from '../../components/Section'

Enzyme.configure({ adapter: new Adapter() })

describe('Section', () => {
  it('should render Board with children', () => {
    const BoardChildren = `
      <div class="row">
        <div class="col-lg-6 text-center">
          <img src="./profile.png" alt="" class="img-fluid  rounded">
        </div>
        <div class="col-lg-6 pt-3 pt-lg-0">
          <div class="text-left">
            <h1>About Me</h1>
          </div>
        </div>
      </div>
    `

    const BoardWrapper = shallow(<Board>{BoardChildren}</Board>)
    const children = BoardWrapper.props().children

    expect(children).toEqual(BoardChildren)
  })

  const SectionChildren = `
      <div class="row">
        <div class="col-lg-6 text-center">
          <img src="./profile.png" alt="" class="img-fluid  rounded">
        </div>
        <div class="col-lg-6 pt-3 pt-lg-0">
          <div class="text-left">
            <h1>About Me</h1>
          </div>
        </div>
      </div>
    `

  it('should render section with custom CSS class', () => {
    const classes = 'bg-white text-dark customClass'
    const wrapper = shallow(<Section id="About" classes={classes}>{SectionChildren}</Section>)

    expect(wrapper.find(Board)).toHaveLength(1)
    expect(wrapper.find(Board).props().children).toEqual(SectionChildren)
    expect(wrapper.hasClass(classes)).toBe(true)
  })

  it('should render section with custom CSS class and Id', () => {
    const classes = 'bg-white text-dark customClass'
    const id = 'myWork'
    const wrapper = shallow(
      <Section classes={classes} id={id}>
        {SectionChildren}
      </Section>
    )

    expect(wrapper.exists(`#${id}`)).toBe(true)

    expect(wrapper.find(Board)).toHaveLength(1)
    expect(wrapper.find(Board).props().children).toEqual(SectionChildren)
    expect(wrapper.hasClass(classes)).toBe(true)
  })

  it('should render section with default CSS class and id when parameter is not passed', () => {
    const wrapper = shallow(<Section id="">{SectionChildren}</Section>)

    expect(wrapper.props().id).toBe('')
    expect(wrapper.find(Board)).toHaveLength(1)
    expect(wrapper.find(Board).props().children).toEqual(SectionChildren)
  })
})
