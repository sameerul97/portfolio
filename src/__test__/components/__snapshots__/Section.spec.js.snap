// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render no text 1`] = `
<p>
  No text
</p>
`;

exports[`Section should render text 1`] = `
<HeaderText
  text="hello world"
/>
`;
